sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriter.java

```diff
@@ -49,4 +50,11 @@ public static CarbonWriterBuilder builder() {
     return new CarbonWriterBuilder();
   }
 
+  public abstract void write() throws IOException;
+
+  /**
+   * To validate and set the list of carbon files.
+   */
+  public abstract void validateAndSetDataFiles(CarbonFile[] dataFiles) throws IOException;
```

Comment:
validation can be handled internally I guess, no need to expose interface

Reply:
only keep SetDataFiles as name

Reply:
Addressed in above comment.